Plugins/NativeServerPlugin/src/conductor.cpp

```diff
@@ -296,21 +290,34 @@ void Conductor::EnsureStreamingUI()
 void Conductor::OnAddStream(rtc::scoped_refptr<webrtc::MediaStreamInterface> stream)
 {
 	LOG(INFO) << __FUNCTION__ << " " << stream->label();
-	main_window_->QueueUIThreadCallback(NEW_STREAM_ADDED, stream.release());
+	if (main_window_->IsWindow())
+	{
+		main_window_->QueueUIThreadCallback(NEW_STREAM_ADDED, stream.release());
```

Comment:
looking below, this just ends up calling `NewStreamAdded` anyway, right? Any reason not to just call it at this point and keep this control flow simple?

Reply:
I think that it's there to make sure that we call **main_window_->StartRemoteRenderer(track)** on UI thread.

Reply:
ohhh, i see. that would make sense